### PR TITLE
Spike3: Retrieve and display code errors without using mongodb

### PR DESCRIPTION
**Pull request by dsellarsnr**

### CODE REVIEW

**Comment by bcanzanella** [comment]
@dsellarsnr are there any specific areas here that would require more attention to detail?

**Comment by dsellarsnr** [comment]
> @dsellarsnr are there any specific areas here that would require more attention to detail?

It's a tough one. CodeErrorNav.tsx and RepositoryAssociator.tsx and CodeError/index.tsx and store/codeErrors/thunks.tsx. The toplevel component is InlineCodemarks.tsx, obviously!

**Review by CalvinAllen — APPROVED** [review]
LGTM*

* +/-99% confidence

File: shared/ui/store/codeErrors/thunks.tsx [hunk 1]
```diff
@@ -98,31 +93,33 @@ export interface CreateCodeErrorError {
...
+			if (response.codeError) {
+				dispatch(addCodeErrors([response.codeError]));
+				dispatch(addStreams([response.stream]));
+				dispatch(addPosts([response.post]));
+			} else if (response.post && response.stream) {
+				// non-mongo mode - enhance the code error with the response
+				const state = getState();
+				const existingCodeError = getCodeError(state.codeErrors, request.errorGuid);
+				if (!existingCodeError) {
+					throw new Error("ohno");
```
Comment (bcanzanella) [inline shared/ui/store/codeErrors/thunks.tsx:111]: can this ever get hit going forward? perhaps a better error message?

**Review by bcanzanella — APPROVED** [review]

**Comment by dsellarsnr** [comment]
@purplexa can you take a look at the last commit e7b83ec41c769a5691341fbc53fb4f66a8231462 for CodeErrorNav? We had a merge conflict.

**Review by purplexa — APPROVED** [review]

**Pull request merged**